### PR TITLE
Introduce the new Outro page for Chapter-1

### LINKED ISSUE #262: Add an outro page for chapter 1
We already have the end page ([here](https://savingsatoshi.com/en/chapters/chapter-1/done)). We want to insert another page the users sees after completing the second challenge that drives the story forward. See the [Figma design](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2374%3A26622&t=1WVTKCVc1kPVBDfc-1) and [export-ready image](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2323%3A25493&t=1WVTKCVc1kPVBDfc-1).

image

### PR DESCRIPTION
**Pull request by shaavan**
Resolves #262

**Screenshot:**

image

### CODE REVIEW

**Comment by shaavan** [comment]
Here's the preview link:

https://saving-satoshi-rjm58k58b-savingsatoshi.vercel.app/en/chapters/chapter-1/outro

**Comment by benalleng** [comment]
tACK!
> Yes, the page background has a different gradient, that's the only change. You can ignore this for now.

I know the color change is shelved for now so other than that it looks good!

**Review by benalleng — COMMENTED** [review]
Looks good! I wonder if making some minor changes to the `Introduction.tsx` template would allow us to not have to create a new template as they are almost identical.

File: ui/chapter/Outro.tsx
Comment (GBKS) [reply ui/chapter/Outro.tsx:42]: Yes, ideally it's the same template.

**Comment by GBKS** [comment]
@ecurrencyhodler has a sharp eye. @shaavan the issue contains a link directly to the export-ready image in Figma that you need here. It should export to 1440x1440px.

**Comment by shaavan** [comment]
>  the issue contains a link directly to the export-ready image in Figma that you need here. It should export to 1440x1440px.

I used the [export-ready image](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2323%3A25493&t=1WVTKCVc1kPVBDfc-1) to create the jpeg. However, I imported it with the `1x` format creating a 720px*720px image. That's why the image was not sharp!

Thanks, @ecurrencyhodler, for noticing the issue, and @GBKS, for pointing the correct dimensions for the export image.

**Comment by GBKS** [comment]
@shaavan did you just press the export button or get the image out another way?

image

**Comment by shaavan** [comment]
Updated from 23ebaa04745e4bae76985aeb327c6fc25f8836ce to e0595bb723ad71f0a668a85302c5a93f241be4fe ([pr266.01](https://github.com/shaavan/saving-satoshi/commits/pr266.01) -> [pr266.02](https://github.com/shaavan/saving-satoshi/commits/pr266.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr266.01..pr266.02))
Addressed @ecurrencyhodler and @benalleng suggestions.

- Introduced a shared UI template between the **Outro** page and **Introduction.tsx.**
- Replaced the outro image with its crispier version.

**Comment by shaavan** [comment]
@GBKS 

>  did you just press the export button or get the image out another way?

By pressing the export button. However, I selected dimensions to be `1x`. Now in the latest update, I have mended the image size/clarity discrepancy.

**Comment by ecurrencyhodler** [comment]
tACK! Nice and clear. Let's get a code review in then merge.
Here's the preview link: https://saving-satoshi-1jevrj33p-saving-satoshi.vercel.app/en/chapters/chapter-1/outro

File: content/lessons/outro/index.tsx
Comment (benalleng) [inline content/lessons/outro/index.tsx:5]: Should we place Outro in the `/done` folder?

**Review by benalleng — CHANGES_REQUESTED** [review]

File: ui/common/TextImageDisplay.tsx
Comment (shaavan) [reply ui/common/TextImageDisplay.tsx:50]: Yes! I observed the modified files in #263. And I saw the Introduction.tsx doesn't display the title and subtitle.
Let me update the template accordingly!

File: content/lessons/outro/index.tsx
Comment (shaavan) [reply content/lessons/outro/index.tsx:5]: The `done` page symbolizes the end of the demo, whereas the `outro` page symbolizes the end of Chapter-1.
So they serve different purposes, and IMO is better to keep them in separate folders.

However, if you think it is better to keep them together, I would love to listen to your take on this.

**Comment by shaavan** [comment]
Updated from e0595bb723ad71f0a668a85302c5a93f241be4fe to 8ff2461 ([pr266.02](https://github.com/shaavan/saving-satoshi/commits/pr266.02) -> [pr266.03](https://github.com/shaavan/saving-satoshi/commits/pr266.03))
Addressed @GBKS and @benalleng comments

Changes:

1. Rebased over the master.
2. Made appropriate changes to the `TextImageDisplay` file to accommodate for latest updates to master
3. Renamed `TextImageDisplay` -> `TextImageLayout`
4. Renamed _chapter-1-outro.jpg_ to _copter-interior-upscaled.jpg_ as per [export-ready image](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=2323%3A25493&t=1WVTKCVc1kPVBDfc-1).

**Comment by shaavan** [comment]
@GBKS

> just to clarify, by export-ready I mean both the file name and export settings are exactly as you should need them. No need to adjust.

The name of the export-ready image is `copter-interior-upscaled 1`, but I have modified it to `copter-interior-upscale`. Couldn't help myself from removing that extra _1_ 😄

@benalleng 

> With the Introduction.tsx and Outro.tsx now using the same template, can Introduction.tsx safely be removed?

I think Introduction.tsx is essential for now because it parses the `chapter` and `intro` metadata along with displaying the UI elements. So removing this file will increase complexity somewhere else.

File: ui/chapter/Outro.tsx
Comment (shaavan) [reply ui/chapter/Outro.tsx]: Resolved in 8ff2461a73207ac7c84677761fa6c236575d86eb

File: ui/chapter/Outro.tsx
Comment (shaavan) [reply ui/chapter/Outro.tsx]: Addressed in 8ff2461a73207ac7c84677761fa6c236575d86eb

File: ui/common/TextImageDisplay.tsx
Comment (shaavan) [reply ui/common/TextImageDisplay.tsx]: Addressed in 8ff2461a73207ac7c84677761fa6c236575d86eb

File: content/lessons/outro/index.tsx
Comment (benalleng) [reply content/lessons/outro/index.tsx:5]: Ok! That makes sense to me!

**Review by benalleng — APPROVED** [review]
It LGTM I just want a final review. I am not at my desk to do a final review though and a want to make sure I am not a blocker.

**Comment by GBKS** [comment]
@shaavan ah, I think I understand now where the misunderstanding comes from. The image is inside a frame. The frame is what you want to export. The frame is necessary sometimes to adjust cropping. 
image

**Review by afterburn — APPROVED** [review]

**Comment by ecurrencyhodler** [comment]
@shaavan looks like there's a merge conflict: "content/lessons/holocat/index.tsx"

**Comment by ecurrencyhodler** [comment]
Okay just as a heads up, once you resolve the remaining item, we can merge. Then we can merge #280 and #279. Then we'll only have one more PR to settle before tomorrow.

**Review by vinayaksh42 — CHANGES_REQUESTED** [review]
@shaavan you will have to rebase your PR, in a recent PR #278 I added a [success page](https://savingsatoshi.com/en/chapters/chapter-1/genesis-4) for the completion of Challenge 1 which is using the `IntroductionLayout.tsx` which you renamed to `TextImageLayout.tsx`. You will have to change the component name for the success page and the location of `holocat` file has been shifted to `content/lesson/genesis/genesis-1`.

The navbar title currently displays `challenge 4` which is a bit wrong as we only have 2 challenges in chapter-1. if possible can you change it to display either `challenge 2` or display only chapter-1 (same as the [`done` page](https://savingsatoshi.com/en/chapters/chapter-1/done)). for making it display `challenge 2` you will have to add it to the `transacting` challenge grouping.
Screenshot 2023-03-12 at 4 16 51 AM

**Comment by shaavan** [comment]
Updated from 8ff2461 to c9a29362a7844b820b59f645686405956d0a28d3 ([pr266.03](https://github.com/shaavan/saving-satoshi/commits/pr266.03) -> [pr266.04](https://github.com/shaavan/saving-satoshi/commits/pr266.04))
Addressed @GBKS @ecurrencyhodler @vinayaksh42 comments

- Rebased over master and resolved merge conflicts.
- Imported the correct Figma frame for creating the image.
- Made the outro page the successor of **challenge 2** (`transacting-4` page) to display the correct challenge number in the navbar.

**Pull request merged**
